inc/saihostif.h

```diff
@@ -875,10 +875,90 @@ typedef enum _sai_hostif_table_entry_channel_type_t
     SAI_HOSTIF_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_LOGICAL_PORT,
 
     /** Receive packets via Linux netdev L3 interface */
-    SAI_HOSTIF_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_L3
+    SAI_HOSTIF_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_L3,
+
+    /** Receive packets via Linux netdev tunnel interface */
+    SAI_HOSTIF_TABLE_ENTRY_CHANNEL_TYPE_NETDEV_TUNNEL
 
 } sai_hostif_table_entry_channel_type_t;
 
+/**
+ * @brief Attribute data for SAI_HOSTIF_TABLE_ENTRY_ATTR_CHANNEL_ENCAP_TYPE
+ */
+typedef enum _sai_hostif_channel_encap_type_t
+{
+    /** Receive packets via GRE/ERSPAN Type 2*/
```

Comment:
Can we just call it ```SAI_HOSTIF_TABLE_CHANNEL_ENCAP_TYPE_GRE``` and add one for IPinIP, then it covers all tunnels:

    SAI_TUNNEL_TYPE_IPINIP,
    SAI_TUNNEL_TYPE_IPINIP_GRE,
    SAI_TUNNEL_TYPE_VXLAN,

Reply:
This is a tunnel to lift packets from the SAI driver to applications. We do not plan to leverage the SAI_TUNNEL infra for this..